Amplify/Categories/DataStore/Model/Internal/Model+Codable.swift

```diff
@@ -76,6 +76,7 @@ extension Model where Self: Codable {
             resolvedEncoder = encoder
         } else {
             resolvedEncoder = JSONEncoder(dateEncodingStrategy: ModelDateFormatting.encodingStrategy)
+            resolvedEncoder.outputFormatting = .sortedKeys
```

Comment:
This looks like a bug fix to ensure the encoded json always has the same key ordering. It's similar to the changes in  https://github.com/aws-amplify/amplify-swift/pull/3057/files#diff-d149ce8efcd8be113dc6cdd44e9e0142ae3853f49cf99a7390c3c50048fd035c that was merged into `main`. Let's keep the code aligned if we can, ie.
```swift
if isKnownUniquelyReferenced(&resolvedEncoder) {
      resolvedEncoder.outputFormatting = .sortedKeys
  }
```

Reply:
I will pull the encorder order changes in a seperate PR.